test/test_transforms.py

```diff
@@ -462,6 +462,15 @@ def test_resize_size_equals_small_edge_size(height, width):
     result = t(img)
     assert max(result.size) == max_size
 
+def test_resize_equal_input_output_sizes(height, width):
+    # Regression test for https://github.com/pytorch/vision/issues/7518
+    height, width = 28, 27
+    img = Image.new("RGB", size=(width, height))
+
+    t = transforms.Resize(((height, width)), antialias=True)
+    result = t(img)
+    assert result is img
```

Comment:
Inplace and no-op behavior is not the same. inplace means that a computation will be executed, but you can choose where the result is stored. no-op means that depending on the input maybe nothing will be done and the input is returned. Otherwise you get a new storage and thus the old one in not modified inplace.

Example for the latter:
```pycon
>>> t = torch.empty((), dtype=torch.float32)
>>> assert t.to(torch.float32) is t
```

That being said, I think it is a reasonable question to ask, if we want no-op functionals. I think this is what @vfdev-5 touched upon in #7518. I don't have a strong opinion here in general. 

For this case in particular, this could be considered a bug fix, since our intention was to support this. I'll dig and find how long this check is broken already.

Reply:
[`nn.Identity`](https://pytorch.org/docs/stable/generated/torch.nn.Identity.html) returns the input itself :)

pytorch interpolate actually will copy the input to the output if input size == output size, here we can also avoid such "unnecessary" copy.

Reply:
We have this no-op behavior for certain transforms in multiple places already. From the top of my head:

- `ConvertDtype`
- All randomly applied transforms, i.e. the ones with the `p` parameter

I think the difference between our transforms and the `nn.Module`'s is that the former is not always just a thin wrapper around a functional, but actually have some functionality that is not available otherwise. Thus, regardless of the way `nn.Module`'s behave, if we even should compare the behavior.

Reply:
`nn.Dropout` also has the [no-op behavior for `p=0`](https://github.com/pytorch/pytorch/blob/ed5395dbef01c796819ae21421b9389d8c89edb9/aten/src/ATen/native/Dropout.cpp#L66-L68):

```python
dropout seems to have no-op behavior though:
from torch.nn.functional import dropout
import torch

a = torch.rand((4, 3, 256, 256))
b = dropout(a, p=0.0, training=True)

torch.testing.assert_close(b, a)
assert b is a
```

Reply:
> I'll dig and find how long this check is broken already.

This short-cut is a fairly recent addition in #6191 and was broken from the get go. Before that, we only had such a check only if the user supplied a single value for size and thus the new size had to be computed at runtime:

https://github.com/pytorch/vision/blob/a5536de95d8e703645e391f4bd885ef489ab35bd/torchvision/transforms/functional_tensor.py#L485-L486

#6191 generalized the parsing of `size` and thus this check was applied to all scenarios not just the one mentioned above.

Technically, #6191 thus already broke BC. This PR would restore it again, but break it for all other scenarios, since they didn't have no-op behavior before. https://github.com/pytorch/vision/pull/6191#pullrequestreview-1016743406 calls for a summary of an offline discussion, but that never happened. @vfdev-5 do you remember if the no-op behavior was discussed?

Reply:
I don't think the `output is [not] input` behaviour is part of our API contract, so I wouldn't say this change is BC-breaking.

This PR provides a (probably minor) speedup for relatively low risk and has a good non-regression test, so I feel like we should be safe to merge.